### DISCUSSION TITLE
RemoteDB Keymaps, Software, Firmware, etc

### DISCUSSION THREAD

**Discussion by rbf121** [GerryDazoo/Slinger#165] [Ideas]
I see folks making their own custom remotes and I was wondering if there is anyway to leverage existing assets for this?  I found some downloads are still accessible which has all of the json, xml, images and spr files leveraged for the various slingplayer on screen remotes.  It also has some SQL queries and DB details which might be useful.  

I have no idea what the structure of the .spr file is but I assume it contains all of the images and animations needed for the on screen remote.  And the json files list the rc code keymaps for each remote.  Is there anyway to use this stuff in your slinger app?  May be worth grabbing it now before it's shutdown.  

Also, this may be a bit overboard but including some other links I found for software and firmware files...

**Remotes**
http://autoupdate.sling.com/remotedb/

**Cable box .bins for IR set up**
http://autoupdate.sling.com/tinybird/ir-sbav/

**Other software downloads:**
http://autoupdate.sling.com/plugin_binary/
http://autoupdate.sling.com/wbsp/
http://autoupdate.sling.com/tinybird/200.00.US.99/

**500 firmware JSON (bin/aes files listed in the json):**
http://mdconfig.sling.com/config/v2/type/ngsb/product/intrepidCbfu/version/01.10.102.json

**M1 firmware JSON (bin files listed in the json):**
http://mdconfig.sling.com/config/v2/type/ngsb/product/apolloCbfu/version/1.2.92.json

350? and others?  I only have an M1 and a 500 but if you can grab your firmware version and use the slingbox device codename, the json might be accessible.  I found a list of codenames used for each device and created the table below but I have no idea if these are truly the latest versions.  (For example, the 500 is listed as 1.10.95 but my installed version is 1.10.102)
        slingboxDetails:{
            "CLASSIC"   : "Slingbox Classic",
            "PRO"       : "Slingbox PRO",
            "AV"        : "Slingbox AV",
            "TUNER"     : "Slingbox TUNER",
            "MODEM"     : "Sling MODEM",
            "SOLO"      : "Slingbox SOLO",
            "PROHD"     : "Slingbox PRO-HD",
            "922"       : "922",
            "HDS600"    : "HDS-600RS",
            "IRIS"      : "Slingbox 120",
            "ROSE"      : "Sling Adapter",
            "INTREPID"  : "Slingbox 500",
            "SABER"     : "Slingbox 350",
            "M1"        : "Slingbox M1",
            "M2"        : "Slingbox M2",
            "UNKNOWN"   : "Slingbox"
        }

SLINGBOX MODEL | FIRMWARE VERSION | Code name for device | Firmware URL
--- | --- | --- | ---
Slingbox 500 | 1.10.95 | INTREPID | http://mdconfig.sling.com/config/v2/type/ngsb/product/intrepidCbfu/version/01.10.095.json
Slingbox M1/M2 | 1.2.092 | APOLLO | http://mdconfig.sling.com/config/v2/type/ngsb/product/apolloCbfu/version/1.2.92.json
Slingbox 350 | 1.5.072 | SABER | http://mdconfig.sling.com/config/v2/type/ngsb/product/saberCbfu/version/1.5.72.json
Slingbox PRO-HD | 2.1.420 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbProHdFwV2_01_420_HwV1_0_0Current.zip
Slingbox SOLO | 2.1.270 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SlingboxSoloFwV2_1_270HwV1_0_3Current.zip
Slingbox Classic | 1.2.16 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPb1_0FwV1_2_16HwV1_0_2Current.zip
Slingbox AV | 1.2.12 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbAVFwV1_2_12HwV1_0_1Current.zip
Slingbox PRO | 1.4.20 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbProFwV1_4_20HwV1_0_3Current.zip
Slingbox TUNER | 1.2.9 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbTunerFwV1_2_9HwV1_0_1Current.zip

Anyone have anything else?

**Comment by flipper2222222** [comment]
I think this might?? be relevant

http://www.hifi-remote.com/forums/dload.php?action=category&cat_id=87

**Comment by GerryDazoo** [comment]
Great stuff. But I'm getting a little busy here. Can someone volunteer to download the JSON and then grab the firmware file names and download them. AFAICT, you need the filename to download the binary for the 350/500, The ProHD files are viewable. Trying to scrap the site only gets you access denied errors.

I'm wondering if there's more than one binary for the 350/500/M1/M2s depending on your hardware version.

**Reply by tom76017** [reply]
I zipped the tar to get it to be loaded in github. The tar has smime files that I don't know how to open.  I think they're encrypted.

[apollo_sbcore_p_1_2_92.zip](https://github.com/GerryDazoo/Slinger/files/9847124/apollo_sbcore_p_1_2_92.zip)

**Reply by flipper2222222** [reply]
> Great stuff. But I'm getting a little busy here. Can someone volunteer to download the JSON and then grab the firmware file names and download them. AFAICT, you need the filename to download the binary for the 350/500, The ProHD files are viewable. Trying to scrap the site only gets you access denied errors.
> 
> I'm wondering if there's more than one binary for the 350/500/M1/M2s depending on your hardware version.

hi Gerry,  if you are requesting for downloading of the files in my reply or rbf121 post? or both?

**Reply by sp222222** [reply]
I scraped all the links rbf121 provided I also have the json, some were already in easy zip file). we are in good shape Gerry.

**Comment by sp222222** [comment]
I've already snagged these but I can start on the others listed here shortly...
Remotes
http://autoupdate.sling.com/remotedb/
3600 files and about 800MB 
It's cool to see 1100 other remote controls :)

**Reply by ncorrare** [reply]
do you have those by any chance? I'm trying to see if I can create an api service for the remotes

**Comment by rbf121** [comment]
Interesting that all of these links still seem to be online, for now.

**Comment by robosmith** [comment]


**Reply by GerryDazoo** [reply]
It's not my remote code. It was submitted by a user. blasco403. He's the guy. 
https://github.com/GerryDazoo/Slinger/discussions/294

**Comment by lepmail** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<body link="#0563C1" vlink="#954F72">

@rbf121's - I added the PRODUCT IDs (found in the console.log files) to your table above.  Hope that is ok. ;)


As pointed out before, these should be latest firmware versions (pre-N9) and the download links are still working.

 **Note on the 500 - 01.10.095.json and 01.10.102.json are identical, except for the signature.**

PRODUCT ID | SLINGBOX MODEL | FIRMWARE VERSION | Code name for device | Firmware URL
-- | -- | -- | -- | --
0 | Slingbox Classic | ? | ? |  
1 | Slingbox PRO | 1.4.20 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbProFwV1_4_20HwV1_0_3Current.zip
2 | Slingbox Classic | ? |   |  
3 | Slingbox AV | 1.2.12 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbAVFwV1_2_12HwV1_0_1Current.zip
4 | Slingbox TUNER | 1.2.9 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbTunerFwV1_2_9HwV1_0_1Current.zip
5 | Slingbox Classic | 1.2.16 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPb1_0FwV1_2_16HwV1_0_2Current.zip
6 | Sling MODEM | ? | ? |  
7 | Slingbox SOLO | 2.1.270 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SlingboxSoloFwV2_1_270HwV1_0_3Current.zip
8 | Slingbox PRO-HD | 2.1.420 | ? | http://autoupdate.sling.com/wbsp/100.07.NA.99/SbPbProHdFwV2_01_420_HwV1_0_0Current.zip
9 | 922 | ? | ? |  
12 | HDS-600RS | ? | ? |  
13 | Slingbox 120 | ? | IRIS ? |  
14 | Sling Adapter | ? | ROSE ? |  
17 | Slingbox 350 | 1.5.072 | SABER | http://mdconfig.sling.com/config/v2/type/ngsb/product/saberCbfu/version/1.5.72.json
18 | Slingbox 500 | 1.10.95 | INTREPID | http://mdconfig.sling.com/config/v2/type/ngsb/product/intrepidCbfu/version/01.10.095.json
32 | Slingbox M1 | 1.2.092 | APOLLO | http://mdconfig.sling.com/config/v2/type/ngsb/product/apolloCbfu/version/1.2.92.json
M2 | Slingbox M2 | 1.2.092 | APOLLO | http://mdconfig.sling.com/config/v2/type/ngsb/product/apolloCbfu/version/1.2.92.json



</body>

</html>

**Comment by ncorrare** [comment]
I found an older slingbox 500 firmware on http://cbfu-prod.slingbox.com/Intrepid/Intrepid_FW_1_8_402v2/intrepid_sbcore_f_p_1_8_402v2.tar

When I blocked a certain IP it seemed to have reverted to some old version of mdconfig I guess (http://mdconfig.sling.com/config/v2/type/ngsb/product/intrepidCbfuRecovery/version/1.3.612.json)
